designer/client/page-edit.js

```diff
@@ -118,15 +123,50 @@ class PageEdit extends React.Component {
       })
   }
 
+  onChangeTitle = e => {
+    const input = e.target
+    const title = input.value
+    this.setState({
+      title: title,
+      generatedPath: this.generatePath(title)
+    })
+  }
+
+  onChangePath = e => {
+    const input = e.target
+    const path = input.value.startsWith('/') ? input.value : `/${input.value}`
+    const sanitizedPath = path.replace(/\s/g, '-')
+    this.setState({
+      path: sanitizedPath
+    })
+  }
+
+  generatePath (title) {
+    let path = toUrl(title)
+    const { data, page } = this.props
+
+    let count = 1
+    while (data.findPage(path) && page.title !== title) {
+      if (count > 1) {
+        path = path.substr(0, path.length - 2)
+      }
+      path = `${path}-${count}`
+      count++
+    }
+    return path
+  }
```

Comment:
not sure why this is a while. can you do a filter and map? in the map callback, index is one of the parameters you can use, so you don't have to keep track of count.

Reply:
I'd have to retrieve all of the pages which contained part of the path, then filter out ones which weren't exact matches (excluding the postfix we're interested in), parse the postfix and find the maximum of those postfixes then add one to it? 
Also I just copied this from page-create so it's existing logic - looks like Max did it originally!